src/app/components/Units.jsx

```diff
@@ -12,12 +12,14 @@ class Units extends React.Component {
   render() {
     const {
       value: propValue, symbol, showSymbol, prefixSymbol, showIcon, iconProps,
-      precision, maxDigits, prefix, suffix, roundingMode, roundingType, ...props
+      precision, abbreviate, maxDigits, prefix, suffix, roundingMode, roundingType, ...props
     } = this.props
     const value = toBigNumber(propValue)
     let expanded = value.toFormat()
-    let shrunk = expanded
-    if (precision) {
+    let shrunk = abbreviate ? abbreviateNumbers(value) : expanded
```

Comment:
Should use same logic for rounding and max digits with abbreviated value. See other comment on `abbreviateNumber` implementation.

```javascript
let shrunk = expanded
let abbrevSuffix
if (abbreviate) {
  let abbreviated = abbreviateNumber(value)
  shrunk = abbreviated.value
  abbrevSuffix = abbreviated.suffix
}
if (precision) {
...
}
if (abbrevSuffix) {
  shrunk = `${shrunk}${abbrevSuffix}`
}
if (symbol) {
...
}
```